### DISCUSSION TITLE
Rescaling pictures from SVG file

### DISCUSSION THREAD

**Discussion by SamHobbsOrg** [wieslawsoltes/Svg.Skia#189] [Q&A]
I have a WPF application that uses SkiaSharp and Svg.Skia. I can load SVG files and show the picture in an XAML skia:SKElement. I cannot find a way to rescale (resize) the images without converting to some other format, such as a bitmap. How do I do that? Is there documentation describing how to do that?

In other words, I want to show an image at a different size, but the entire image, not cropped. It is my understanding that that is a fundamental feature of SVG.

**Comment by wieslawsoltes** [comment]
You want something like Svg control from Avalonia ported. WPF is not supported.

**Comment by SamHobbsOrg** [comment]
Thank you, you are the developer, you are the expert.

I should have been more clear, for those not familiar with WPF. This is a .Net application using C#.

I am using Svg.Skia to load SVG files and SkiaSharp to draw it. Therefore I think that WPF is irrelevant to my use of Svg.Skia. If Svg.Skia will not work then I must accept that but I want to ensure that the requirements are understood.

I have heard of Avalonia and I have considered trying it. For this project, if .Net MAUI will satisfy requirements then that is the alternative.

**Comment by SamHobbsOrg** [comment]
I have learned that probably I just need to change the width and height in the XML to get a new size. And probably viewBox, I am a beginner. Modifications such as that are of course very common, I just need to figure out how to do that.

**Comment by SamHobbsOrg** [comment]
The answer is to use a `SKMatrix` as in:

`SkiaSharp.SKMatrix matrix = SKMatrix.CreateScale(.5F, .5F);`

Then instead of `canvas.DrawPicture(svg.Picture);` we can do `canvas.DrawPicture(svg.Picture, ref matrix);`.

That works in WPF.
